docs/tutorial-run-solver.md

```diff
@@ -0,0 +1,176 @@
+---
+id: tutorial-solver
+title: Quickstart - Running your own Solver
+sidebar_label: Quickstart
+---
+
+#
+
+# Contents
+
+1. [Installation](#installing-driver-tools)
+2. [Building The Driver](#building-the-driver)
+3. [Configuring Environment](#configuring-environment-variables)
+4. [Acquiring the Orderbook](#acquiring-the-orderbook)
+5. [Run the Solver](#run-the-solver)
+6. [Advanced Configuration](#advanced-configuration)
+   1. [Orderbook Filter](#Configuring-an-Orderbook-Filter)
+
+---
+
+## Installing Driver Tools
+
+Clone the dex-services repository, and acquire the project requirements
+
+```sh
+git clone git@github.com:gnosis/dex-services.git
+```
+
+We demonstrate how to run the open-solver. Once comfortable, the open-solver can be adapted to suit your own needs from the open-solver repo
+
+```sh
+git clone git@github.com:gnosis/dex-solver.git
+```
+
+## Building The Driver
+
+Build the driver (with open-solver dockerhub image) from within the root directory of `dex-services` repo. This command builds the driver while simultanesouly baking the open solver into the the build.
+
+```sh
+docker-compose build --build-arg SOLVER_BASE=gnosispm/dex-open-solver:master stablex-debug
+```
+
+Note that, if `SOLVER_BASE` is not specified, the driver's own internal `naive-solver` will be used. Alternatively, if you have your own personalized solver, this image would need to be specified during the build.
+We will see that specifying solver type again at runtime is also expected and this may seem counterintuitive.
+However, the argument used at runtime is meant to specify potentially different implementations from within the existing solver build.
+For example, you may have a `standard-solver` which matches orders according to one algorithm and another, say `fallback-solver` which makes use of a fixed list of currated tokens or simply operates in a different way.
+
+## Configuring Environment Variables
+
+The following three environment variables are required when running the dex-services project.
+Note that the account corresponding to the private key should be funded with sufficient ETH to pay gas for solution submission.
+
+```sh
+export PRIVATE_KEY=<YOUR_PRIVATE_KEY>
+export ETHEREUM_NODE_URL=https://mainnet.infura.io/v3/<YOUR_INFURA_KEY>
+export NETWORK_ID=<YOUR_PREFERED_NETWORK_ID>
+```
+
+By placing these values into an `.env` file and sourcing this file, you can avoid adding them to as runtime arguments on every execution.
```

Comment:
This is not the s for me when I mounted the project inside the container.

Reply:
Ok, then this is most likely what `$(PWD)` is doing. What do you see if you run `echo $(PWD)`?

Reply:
```
➜  dex-docs git:(integration_tutorial) echo $(PWD)
/Users/admin/Projects/gnosis/dex-docs
➜  dex-docs git:(integration_tutorial) echo $PWD
/Users/admin/Projects/gnosis/dex-docs
```

I guess they are both the same. We could ask @fleupold if he had something particular in mind at the time this was included in the README.

Reply:
In Bash `$(command)` is substituted with the output of `command`, so that you can do something like:
```
$ echo "The current timestamp is $(date +%s)."
The current timestamp is 1599823115.
```
So the reason why `$(PWD)` is failing for me is that `PWD` is not an existing command. Assuming that the syntax doesn't change in your terminal, then it'd mean that there is a `PWD` command available in MacOS, in case you should be able to run it just by typing `PWD` in the console. (Alternatively, maybe commands are not case sensitive? `pwd` would be a valid command for me.)
By the way, `$PWD` (or equivalent but safer, `${PWD}`) works also for me, so we could use that for compatibility.

Reply:
`$(PWD)` is a ZSH thing, I think we should stick to Bash for compatibility.

Reply:
In case you are curious: I got hold of a Mac and was able to test this. Apparently `PWD` is indeed a command in MacOs, so `$()` is working in `zsh` as expected for Bash. Interestingly, `pwd` is a "shell builtin" while `PWD` is a standalone executable.
In my Linux machine, `PWD` doesn't exist and `pwd` is a standalone executable.